packages/pwa-kit-create-app/assets/bootstrap/js/jest.config.js

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const base = require('@salesforce/pwa-kit-dev/configs/jest/jest.config.js')
+
+module.exports = {
+    ...base,
+    // Don't ignore @salesforce/retail-react-app
```

Comment:
it's more important to have the "why"s than "how"s in code comment.
```suggestion
    // To support extensibility, jest needs to transform the underlying templates/extensions
```

Reply:
what happens if the project wants to extend from another project or have multiple extensions

if the package.json looks like

```json
{
  ccExtensibility: ['@salesforce/retail-react-app', 'myCustomExtensionA']
}
```

Does that mean they'll need to update this jest config to include all extensions? How can we build an API that does that for the users? 🤔

Reply:
for example, can the `base` jest.config.js be smart enough to automatically build `transformIgnorePatterns` that includes all extensions?

Reply:
From what I understand, all the extensions will live under the `@salesforce/extensions` directory in the `node_modules`, so when we implement that change we can just update the jest config to transform the files in that directory

Reply:
does that mean we'll run babel to transpile all node_modules dependencies? I wonder if that has negative impact on performance, do you want to give it a try and see if it works?

Reply:
No, if we wanted to transform all the dependencies inside `node_modules`, we would set `transformIgnorePatterns` to an empty array

```javascript
module.exports = {
    ...base,
    // this means ignore nothing, meaning transform everything, including node_modules
    transformIgnorePatterns: [],
    moduleNameMapper: {
        ...base.moduleNameMapper,
        '^is-what$': '<rootDir>/node_modules/is-what/dist/cjs/index.cjs',
        '^copy-anything$': '<rootDir>/node_modules/copy-anything/dist/cjs/index.cjs'
    }
}
```

Here are the performance results of how long it takes to run `npm run test` in the demo project:

**Transform only @salesforce/retail-react-app (how it is currently set up)**
- First time: 6.158 s
- Second time: 2.899 s

**Transform all node modules**
- First time: 28.537 s
- Second time: 3.973 s

Reply:
thanks for trying it out, yea i think we definitely don't want to transform all node modules, it's too slow.